### PR TITLE
bpo-30502: Fix handling of long oids

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

### CODE REVIEW

**Comment by tiran** [comment]
Similar to https://github.com/pyca/cryptography/pull/3612

**Review by serhiy-storchaka — APPROVED** [review]
LGTM, but I suggest the patch that don't duplicate the code (see on the tracker).

File: Modules/_ssl.c [hunk 1]
```diff
@@ -753,19 +753,35 @@ _ssl__SSLSocket_do_handshake_impl(PySSLSocket *self)
...
+    if (buflen > X509_NAME_MAXLEN - 1) {
+        /* make OBJ_obj2txt() calculate the required buflen */
+        buflen = OBJ_obj2txt(NULL, 0, name, 0);
+        /* allocate len + 1 for terminating NULL byte */
+        namebuf = PyMem_Malloc(buflen + 1);
+        if (namebuf == NULL) {
+            PyErr_NoMemory();
+            goto fail;
+        }
+        buflen = OBJ_obj2txt(namebuf, buflen + 1, name, 0);
```
Comment (vstinner) [inline Modules/_ssl.c:779]: Do we need that the new buflen is equal to the previous buflen?

**Comment by serhiy-storchaka** [comment]
An alternate patch is provided as #2909.

**Comment by tiran** [comment]
Closing in favor of #2909.

**Pull request closed without merging**